tiledb/sm/filesystem/win.cc

```diff
@@ -586,10 +581,13 @@ Status Win::write_at(
   // file handle. Instead, we use the OVERLAPPED struct to specify an offset at
   // which to write. Note that the file handle does not have to be opened in
   // "overlapped" mode (i.e. async writes) to do this.
-  unsigned long bytes_written = 0;
   uint64_t byte_idx = 0;
   const char* byte_buffer = reinterpret_cast<const char*>(buffer);
-  while (buffer_size > constants::max_write_bytes) {
+  while (buffer_size > 0) {
+    DWORD bytes_to_write = buffer_size > std::numeric_limits<DWORD>::max() ?
```

Comment:
Consider either renaming buffer_size to something else reflecting 'remaining_bytes_to_write' or introduce a local variable reflecting that idea to use in place of 'buffer_size'.